### ISSUE TITLE
Failure to install xerces-c 3.1.4

### ISSUE THREAD

**Issue by paercebal** [Microsoft/vcpkg#1726]
I've tried to reinstall VCPKG, and it failed on Xerces 3.1.4 (and thus, Xalan, which depends on it).

Looking around:
1 - My previous install of VCPKG still has Xerces 3.1.4 (from May, 2017)
2 - I just update my current install of VCPKG (git pull, etc.)
3 - The URL is indeed now invalid : http://www-us.apache.org/dist//xerces/c/3/sources/xerces-c-3.1.4.zip
4 - Apache seemed to have delivered Xerces 3.2.0

I suspect they removed 3.1.4 the moment 3.2.0 went out, thus breaking those who depended on it.
I can't find any reliable link to old zips for Xerces. The only thing I found was a github repository: https://github.com/apache/xerces-c/ but I'm not sure this is the main repository. I'm wondering if this is not a simple snapshot of major versions.

Complete logs:

```
D:\__pkg\vcpkg>vcpkg install xerces-c:x64-windows 
The following packages will be built and installed:
    xerces-c:x64-windows
Building package xerces-c:x64-windows... 
-- CURRENT_INSTALLED_DIR=D:/__pkg/vcpkg/installed/x64-windows
-- DOWNLOADS=D:/__pkg/vcpkg/downloads
-- CURRENT_PACKAGES_DIR=D:/__pkg/vcpkg/packages/xerces-c_x64-windows
-- CURRENT_BUILDTREES_DIR=D:/__pkg/vcpkg/buildtrees/xerces-c
-- CURRENT_PORT_DIR=D:/__pkg/vcpkg/ports/xerces-c/.
-- Downloading http://www-us.apache.org/dist//xerces/c/3/sources/xerces-c-3.1.4.zip...
-- Downloading http://www-us.apache.org/dist//xerces/c/3/sources/xerces-c-3.1.4.zip... Failed. Status: 22;"HTTP response code said error"
CMake Error at D:/__pkg/vcpkg/scripts/cmake/vcpkg_download_distfile.cmake:82 (message):
  

      Failed to download file.
      Add mirrors or submit an issue at https://github.com/Microsoft/vcpkg/issues

Call Stack (most recent call first):
  D:/__pkg/vcpkg/ports/xerces-c/portfile.cmake:22 (vcpkg_download_distfile)
  D:/__pkg/vcpkg/scripts/ports.cmake:72 (include)


Error: Building package xerces-c:x64-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: xerces-c:x64-windows
  Vcpkg version: 0.0.85-2017-08-28-28e581599e263e3f0b021ad5b6914d80fa9640f8
```

**Comment by codicodi** [comment]
We should probably use github mirror: https://github.com/apache/xerces-c

**Comment by paercebal** [comment]
Looking at how they release each version there (https://github.com/apache/xerces-c/releases), I agree.

**Comment by paercebal** [comment]
Ok, I updated the port file to use github instead (it was guess work, as I'm not CMake fluent).
It works for x86/x64-windows, but not for x86-uwp:

```
D:\_dev\github.com.paercebal\vcpkg>vcpkg install xerces-c:x86-uwp
The following packages will be built and installed:
    xerces-c:x86-uwp
Building package xerces-c:x86-uwp...
-- CURRENT_INSTALLED_DIR=D:/_dev/github.com.paercebal/vcpkg/installed/x86-uwp
-- DOWNLOADS=D:/_dev/github.com.paercebal/vcpkg/downloads
-- CURRENT_PACKAGES_DIR=D:/_dev/github.com.paercebal/vcpkg/packages/xerces-c_x86-uwp
-- CURRENT_BUILDTREES_DIR=D:/_dev/github.com.paercebal/vcpkg/buildtrees/xerces-c
-- CURRENT_PORT_DIR=D:/_dev/github.com.paercebal/vcpkg/ports/xerces-c/.
-- Using cached D:/_dev/github.com.paercebal/vcpkg/downloads/Xerces-C_3_1_4.zip
-- Testing integrity of cached file...
-- Testing integrity of cached file... OK
-- Extracting done
-- Building D:/_dev/github.com.paercebal/vcpkg/buildtrees/xerces-c/src/xerces-c-Xerces-C_3_1_4/projects/Win32/VC14/xerces-all/xercesLib/xercesLib.vcxproj for Release
-- Building D:/_dev/github.com.paercebal/vcpkg/buildtrees/xerces-c/src/xerces-c-Xerces-C_3_1_4/projects/Win32/VC14/xerces-all/xercesLib/xercesLib.vcxproj for Debug
-- Performing post-build validation
The following DLLs do not have the App Container bit set:

    D:/_dev/github.com.paercebal/vcpkg/packages/xerces-c_x86-uwp/debug/bin/xerces-c_3_1D.dll
    D:/_dev/github.com.paercebal/vcpkg/packages/xerces-c_x86-uwp/bin/xerces-c_3_1.dll

This bit is required for Windows Store apps.
Found 1 error(s). Please correct the portfile:
    D:\_dev\github.com.paercebal\vcpkg\ports\xerces-c\portfile.cmake
-- Performing post-build validation done
Error: Building package xerces-c:x86-uwp failed with: POST_BUILD_CHECKS_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: xerces-c:x86-uwp
  Vcpkg version: 0.0.85-2017-08-28-28e581599e263e3f0b021ad5b6914d80fa9640f8
```
As I don't use "uwp", I have no idea what this means. A quick search led to a fix that is unrelated: https://github.com/Microsoft/vcpkg/issues/111

Bonus: Xalan seems to still work, but I believe a migration to github would also be a good idea.

**Comment by ras0219-msft** [comment]
It's highly unlikely that this is a regression -- I suspect the current Xerces port doesn't work for uwp either.

I'd be happy to accept a PR changing the upstream source to GitHub!

**Comment by paercebal** [comment]
Let's go!
:-)

**Comment by alexkaratarakis** [comment]
Fixed by #1741 :).
